### DISCUSSION TITLE
Can we include variation from noise into variation from prior?

### DISCUSSION THREAD

**Discussion by hyunjimoon** [Data4DM/BayesSD#70] [perspectives; sense-making]
If we were to imagine the last M number of effective MCMC draws, its variation would be the result of added variations from
a. prior
b. process noise
c. measure noise
d. MCMC sampling

@hyunjimoon thinks b, c can be included in a.
@tomfid may I ask your opinion?

**Comment by tomfid** [comment]
Does "effective" mean "accepted" here? 

I think the process noise is potentially tricky, so I'll ignore that for the moment.

It seems like broader priors, or more measurement error in the data, should produce more variation in theta.

If the sampler is performing well, then shouldn't the variation in M draws be reasonably constant? OTOH if the sampler is underperforming (stuck in a corner, or creeping slowly), variation could be nonstationary unless M is very large?

If "included in a" means that part of theta is a parameter describing the scale of measurement error that produced y, I agree.

**Reply by hyunjimoon** [reply]
> Does "effective" mean "accepted" here?

We need a unified terms for here as HMC is not have accept-reject. Hazhir and I agreed that for Stan's M, effective sample size from the [this](https://github.com/Data4DM/stanify#map-) table below is usable for now.
<img width="913" alt="image" src="https://user-images.githubusercontent.com/30194633/198705851-2d7d176f-857f-473a-b511-4a23122bf62e.png">

**Reply by hyunjimoon** [reply]
> process noise is potentially tricky

From implementation perspective, driving data and process noise are treated equally (both driving data and random variable's one realization of value are passed through stan data block in [this line 7,8](https://streaklinks.com/BPyb3xR-fhAeXlvuQwvwb6Lc/https%3A%2F%2Fgithub.com%2FData4DM%2Fstanify%2Fblob%2Fmain%2Fstan_files%2Fprey_predator_1020%2Fprey_predator_1020_draws2data.stan%23L7?email=hyunji.moonb%40gmail.com) of stancode). This is the only way to implement the random variable concept of SDE for now (and for the future as well, I think).  [This post explains](https://streaklinks.com/BPyb3w8qKlJv6LT2cwq1qtNW/https%3A%2F%2Fdiscourse.mc-stan.org%2Ft%2Fhow-to-solve-stochastic-differential-equations-in-stan%2F1162%2F3?email=hyunji.moonb%40gmail.com) Stan cannot solve SDE and introduces Mike Bentancourt's remedy. 

_We can’t do SDEs per se, but if you can transform the problem into a Gaussian process with an appropriate kernel then you can tackle it from the GP perspective. Also, you can do the INLA trick of discretizing the domain of the SDE and modeling it with random effects – that can do very well in many applications_

The above comment reminds me of how you explained two ways to model demand data (driving data and parameterizing it with correlation and scale parameter). Viewing alpha and rho from [Mike's Gaussian process writing](https://streaklinks.com/BPyb3w8Xl8iGKE_HogqeODiL/https%3A%2F%2Fbetanalpha.github.io%2Fassets%2Fcase_studies%2Fgaussian_processes.html%2313_Covariance_Functions_and_You?email=hyunji.moonb%40gmail.com) as process noise scale and correlation time of process noise, I can't distinguish driving data and process noise well.. (let alone their equivalent implementation format).

**Reply by tomfid** [reply]
I think the relevant difference between driving data and process noise is that process noise is not observed directly, so you can only infer its distribution?

**Reply by hyunjimoon** [reply]
Do you think, between using one realization of data VS using one noise seed's uniform random series, there is much difference in degrees of clamping?
